tests/apps/core/test_api_rooms.py

```diff
@@ -0,0 +1,232 @@
+
+
+class RoomsApiTestCase(APITestCase):
+    """Test requests on magnify's core app API endpoints."""
+
+    def test_api_rooms_list_anonymous(self):
+        """Anonymous users should not be allowed to list rooms."""
+        RoomFactory.create_batch(2)
+        response = self.client.get("/api/rooms/")
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(
+            response.json(), {"detail": "Authentication credentials were not provided."}
+        )
+
+    def test_api_rooms_list_authenticated(self):
+        """Authenticated users should be able to list rooms in which they are administrators."""
+        user = UserFactory()
+        rooms = RoomFactory.create_batch(2, administrators=[user])
+        RoomFactory()
+        jwt_token = AccessToken.for_user(user)
+
+        response = self.client.get(
+            "/api/rooms/", HTTP_AUTHORIZATION=f"Bearer {jwt_token}"
+        )
+
+        self.assertEqual(response.status_code, 200)
+        results = response.json()
+        self.assertEqual(len(results), 2)
+        expected_ids = {r.id for r in rooms}
+        results_id = {r["id"] for r in results}
+        self.assertEqual(expected_ids, results_id)
+
+    def test_api_rooms_retrieve_anonymous(self):
+        """Anonymous users should not be allowed to retrieve a room."""
+        room = RoomFactory()
+        response = self.client.get(f"/api/rooms/{room.id}/")
+
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(
+            response.json(), {"detail": "Authentication credentials were not provided."}
+        )
+
+    @mock.patch(
+        "magnify.apps.core.serializers.rooms.generate_token", return_value="the token"
+    )
+    def test_api_rooms_retrieve_authenticated(self, mock_token):
+        """Authenticated users should be allowed to retrieve a room and get a guest token."""
+        admin = UserFactory()
+        room = RoomFactory(administrators=[admin])
+        user = UserFactory()
+        jwt_token = AccessToken.for_user(user)
+
+        response = self.client.get(
+            f"/api/rooms/{room.id}/", HTTP_AUTHORIZATION=f"Bearer {jwt_token}"
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.json(),
+            {
+                "administrators": [admin.id],
+                "id": room.id,
+                "name": room.name,
+                "slug": room.slug,
+                "token": "the token",
+            },
+        )
+        mock_token.assert_called_once_with(user, room.slug)
+
+    @mock.patch(
+        "magnify.apps.core.serializers.rooms.generate_token", return_value="the token"
+    )
+    def test_api_rooms_retrieve_administrator(self, mock_token):
+        """Administrators of a room should get an admin token."""
+        admin = UserFactory()
+        room = RoomFactory(administrators=[admin])
+        user = UserFactory()
+        jwt_token = AccessToken.for_user(user)
```

Comment:
Should not be the admin here ?

Reply:
Well spotted :clap:  This test is not needed anymore since I finally chose to mock the token generation.